### DISCUSSION TITLE
link file displayed on two windows, when scroll file in one window, the file on other window scrolls correspandently

### DISCUSSION THREAD

**Discussion by Lioncat-M** [sumatrapdfreader/sumatrapdf#2822] [Ideas]
The typical scence for this is a translated file and the original file, user read the translated file, sometime need to refer to original file for better understanding.

**Comment by GitHubRulesOK** [comment]
Generally a feature of windows operating system e.g. scrolling two apps is not practical for many reasons such as security/interference/scaling.

Say you have a terse language and a verbose language side by side
then your requirement is different scroll rates, yes I know it could be relative all page # L to sync with all page # R but how can the mouse or key know which window belongs to one keyboard or touchpad.

 There are generically two workarounds.

One is a 3rd party key/mouse/pad app controls the two windows so a scroll or page down is applied to both and the simplest way is using AutoHotKey to apply control to two custom windows. There was a lengthy discussion about using native DDE in https://github.com/sumatrapdfreader/sumatrapdf/issues/1398


The other is SumatraPDF in one window synchronizes a presentation in another window and that was covered some time ago on old forum but its fiddly to try part pages so best limited to same page or plus/minus one page at a time. again a lengthy discussion but a viable workaround was here https://forum.sumatrapdfreader.org/t/current-page-number-of-the-page-displayed-for-presentations/4144

**Reply by Lioncat-M** [reply]
Thank you for the quick reply and substantially sugestion! 
I'm asking for two windowed SumatraPDF to sync linked files, as I checked in win10 Sumatra has two threads, I guess the two are to operate 2 windows,  so should has no security issue as them are  under one Sumatra process. The best sync way I thought is  two well bookmarked files, window scroll rate based on correspondent bookmark section words count. The worst sync could br two files without any book marks, we can roughly sync on file's total words count. Thanks again!

**Reply by GitHubRulesOK** [reply]
words are not a function of PDF it is `Document has page(s) has character(s) or not` at different scaled heights and lengths thus comparing letters in 2 pages is meaningless.

The closest is window1 Y100 = window2 Y100, e.g. scroll both by same scale. However what if one is zoom in and one is zoom out then its unwieldy to try to manage. To tie two windows together is of limited value, except for dual screen page by page presentation, since otherwise there are too many different interrelated issues.

I have tried different ways to run tiled windows, and best was two frames side by side in a browser where the heights are constrained to identical, however that is only possible in a few remaining candidates
![image](https://user-images.githubusercontent.com/12754913/180787140-55b9e224-e9e6-4116-be16-9c4f54b0fd7a.png)

**Reply by Lioncat-M** [reply]
Seems not an easy job. Thank you!
